docs/details/other-components/xml.rst

```diff
@@ -0,0 +1,301 @@
+but rather compiled into the application. The main characteristics of widgets are:
+
+- Similar to LVGL's built-in widgets.
+- Built from classes.
+- Have a large API with set/get/add/etc. functions.
+- Support "internal widgets" (e.g., tabview's tabs, dropdown's list).
+- Have custom and complex logic inside.
+- Cannot be loaded from XML at runtime because custom code cannot be loaded.
+
+Components are built from other components and widgets and can be loaded at runtime.
+The main characteristics of components are:
+
+- Built from widgets or other components.
+- Can be used for styling widgets.
+- Can contain widgets or other components.
+- Cannot have custom C code.
+- Can be loaded from XML at runtime as they describe only the visuals.
+
+Components
+----------
+
+Overview
+~~~~~~~~
+
+In light of the above, only components can be loaded from XML.
+An example of a ``my_button`` component looks like this:
+
+.. code-block:: xml
+
+    <component>
+        <consts>
+            <px name="size" value="100"/>
+            <color name="orange" value="0xffa020"/>
+        </consts>
+
+        <params>
+            <string name="btn_text"/>
+        </params>
+
+        <styles>
+            <style name="blue" bg_color="0x0000ff" radius="2"/>
+            <style name="red" bg_color="0xff0000"/>
+        </styles>
+
+        <view extends="lv_button" width="#size" styles="blue red:pressed">
+            <my_h3 text="$btn_text" align="center" color="#orange" style_text_color:checked="0x00ff00"/>
+        </view>
+    </component>
+
+- ``<component>``: The root element.
+- ``<consts>``: Constants with ``px`` or ``color`` types.
+Constants can later be referenced as ``#name``.
+- ``<params>``: Parameters with ``int``, ``px``, ``string``, ``color``, or ``style`` types.
+Parameters can later be referenced as ``$name``.
+- ``<styles>``: ``<style>`` properties can be defined with names and properties.
+- ``<view>``: Describes how the component looks. Can reference constants, parameters, and styles.
+
+Naming conventions:
+
+- A standard XML syntax is used.
+- Lowercase letters with ``_`` separation are used for attribute names.
+- The usual variable name rules apply for attribute and tag names: only letters, numbers, `'_'` and can't start with a number.
+- LVGL API is followed as much as possible, e.g., ``align="center"``, ``bg_color="0xff0000"``.
+- ``params`` can be referenced with ``$``
+- ``consts`` can be referenced with ``#``
+- ``styles`` can be attached to states and/or parts like ``styles="red blue:pressed green:focused:scrollbar"``
+- Local styles can be used like  ``<lv_label style_text_color="0xff0000" style_text_color:checked="0x00ff00" ``
+Usage
+~~~~~
+
+Once a component is created (e.g., ``my_button``), it can be registered by calling either:
+
+- ``lv_xml_component_register_from_file("A:lvgl/examples/others/xml/my_button.xml");``
+- ``lv_xml_component_register_from_data("my_button", xml_data_of_my_button);``
```

Comment:
I think the idea of lv_image_set_src(void*) can apply here too, so `lv_xml_component_register` is enough.

Name of `lv_xml_register_component` looks more similar to `lv_image_set_src`.

Reply:
In case of images `lv_image_dsc_t` is a special struct where the first byte is a magic number to distinguish it from a string. It's not true here as `xml_data_of_my_button` is just a simple array.

Reply:
It's still possible to judge if the first argument(string) is ended with ".xml".

Reply:
I can't check the string length because it might be a pointer to a some data too.

Reply:
I mean the first argument, in your example, they are both string but one of them is ended with .xml. :)